src/trio/_repl.py

```diff
@@ -10,14 +14,19 @@
 
 
 class TrioInteractiveConsole(InteractiveConsole):
-    def __init__(self, repl_locals=None):
-        super().__init__(repl_locals)
+    def __init__(self, repl_locals: dict[str, object] | None = None):
+        super().__init__(locals=repl_locals)
         self.compile.compiler.flags |= ast.PyCF_ALLOW_TOP_LEVEL_AWAIT
 
-    def runcode(self, code):
-        async def _runcode_in_trio():
+    def runcode(self, code: types.CodeType) -> None:
+        async def _runcode_in_trio() -> BaseException | None:
+            # code.InteractiveInterpreter defines locals as Mapping[str, Any]
+            # However FunctionType expects a dict. We know our copy of
+            # locals will be a dict due to the annotation on repl_locals in __init__
+            # so the cast is safe.
```

Comment:
You might be able to put a variable declaration and avoid this? Not sure how to phrase that, but like:

```py
class TrioInteractiveConsole(InteractiveConsole):
    locals: dict[str, object] | None
```

Reply:
Yes, that worked. Must nicer than having to cast